content/blog/2017/2017-01-12-scm-api-2.0-release.adoc

```diff
@@ -0,0 +1,230 @@
++
+[WARNING]
+====
+If you upgrade to Branch API 2.0.x and you have either the GitHub Branch Source or the BitBucket Branch Source plugins *and* you do not upgrade those instances to the 2.0.x line then your Jenkins instance will fail to start-up correctly.
+
+The solution is just to upgrade the GitHub Branch Source or the BitBucket Branch Source plugin (as appropriate) to the 2.0.x line.
+====
+
+Git Plugin::
+The new releases of this plugin are both API and on-disk compatible with plugins compiled against the previous releases.
++
+The 2.0.x lines of both the GitHub Branch Source and BitBucket Branch Source plugins require that you upgrade your Git Plugin to one of the versions that supports SCM API 2.0.x.
+In general, the required upgrade will be performed automatically when you upgrade your GitHub Branch Source and BitBucket Branch Source plugins.
+
+Mercurial Plugin::
+The new release of this plugin is both API and on-disk compatible with plugins compiled against the previous releases.
++
+The 2.0.x line of the BitBucket Branch Source plugins require that you upgrade your Mercurial Plugin to the 2.0.x line.
+In general, the required upgrade will be performed automatically when you upgrade your  BitBucket Branch Source plugins.
+
+BitBucket Branch Source Plugin::
+The 2.0.x line of this plugin makes on-disk file format changes that mean you will be unable to roll back to the 1.x line after an upgrade without restoring the old data files from a back-up.
+
+GitHub Branch Source Plugin::
+The 2.0.x line of this plugin makes on-disk file format changes that mean you will be unable to roll back to the 1.x line after an upgrade without restoring the old data files from a back-up.
++
+[WARNING]
+====
+If you upgrade to GitHub Branch Source 2.0.x and you have the GitHub Organization Folders plugin installed, you must upgrade that plugin to the tombstone release.
+====
+
+GitHub Organization Folders Plugin::
+The functionality of this plugin has been migrated to the GitHub Branch Source plugin.
+You will need to upgrade to the tombstone release in order to ensure all the data has been migrated to the classes in the GitHub Branch Source plugin.
++
+[TIP]
+====
+Once you have upgraded to the tombstone version and all GitHub Organization Folders have had a full scan completed successfully, you can disable and uninstall the GitHub Organization Folders plugin.
+There will be no more releases of this plugin after the tombstone.
+The tombstone is only required for data migration.
+====
+
+[[tldr]]
+== Summary for busy Jenkins Administrators
+
+Upgrading should make multi-branch projects much better.
+When you are ready to upgrade you must ensure that you upgrade all the required plugins.
+If you miss some, just upgrade them and restart to fix the issue.
+
+TIP: Always take a backup of your `JENKINS_HOME` before upgrading any plugins.
```

Comment:
Put this tip at the top as well (with the other two key points).   Telling people twice to backup before upgrading is better than them not seeing it.